src/components/CartCard.jsx

```diff
@@ -0,0 +1,101 @@
+//ToDo Rename this file, confused in the structure a bit
+//ToDo make UI Look good. 
+
+import axios from "axios";
+import { useState } from "react";
+import useMainState from "../contexts/MainContextProvider";
+import ENUMS from '../utils/enums';
+
+const { SET_CART, SET_WISHLIST } = ENUMS;
+
+const SingleProductCart = ({ product, quantity }) => {
+    // 0 ->  not done, 1 -> loading, 2-> successful, -1 -> failed
+    const [addingToWishlist, setAddingToWishlist] = useState(0);
+    const [updatingCart, setUpdatingCart] = useState(false);
+    const {dispatch} = useMainState();
+
+
+    const updateCart = (quantity) => {
+        setUpdatingCart(true);
+        axios.post(`${process.env.REACT_APP_SERVER_URL}/cart/`, {
+                productId: product._id,
+                quantity
+            }).then((res) => {
+                dispatch({type: SET_CART, data: res.data});
+            })
+            .catch((e) => {
+                console.log(e);
+                //show error here.
+            })
+            .finally(() => {
+                setUpdatingCart(false);
+            });
+    };
+
+    const addToWishlist = async () => {
+        setAddingToWishlist(1);
+        try{
+            const {data : cart} = await axios.delete(`${process.env.REACT_APP_SERVER_URL}/cart/`, {
+                data : {productId: product._id}
+            });
+            const {data : wishlist} = await axios.post(`${process.env.REACT_APP_SERVER_URL}/wishlist/`, {
+                productId: product._id,
+            });
+            dispatch({type: SET_WISHLIST, data: wishlist});
+            dispatch({type: SET_CART, data: cart});
+            setAddingToWishlist(2);
+        } 
+        catch (err) {
+            console.log(err);
+            setAddingToWishlist(-1);
```

Comment:
0,1,2,-1, does not make sense to a person not worked on this. Please create global variables like `const SUCCESS = 0` and then use SUCCESS here, so that it's readable